db/chain/chain.go

```diff
@@ -220,23 +220,35 @@ func (ec *ExpresionChain) Delete() *ExpresionChain {
 	return ec
 }
 
-// ConflictAction represents a possible conflict resolution action.
-type ConflictAction string
+// Conflict will add a "ON CONFLICT" clause at the end of the query if the main operation
+// is an INSERT.
+func (ec *ExpresionChain) OnConflict(clause func(*OnConflict)) *ExpresionChain {
+	if ec.conflict != nil {
+		panic("only 1 ON CONFLICT clause can be associated per statement")
```

Comment:
Don't panic®
Libraries should never panic.
Either make this a non chainable method, which returns err and whatever else or make it override previous conflict.